haystack/utils/object_inspection.py

```diff
@@ -0,0 +1,24 @@
+import inspect
+from typing import Callable
+
+
+def get_parameter_info(method: Callable):
```

Comment:
I would move this in `Pipeline` as a private method, it's probably going to be used only there.

Also it would be better passing the Component instance directly, you can access the inputs info from the field `__haystack_input__`. That is an instance of [`Sockets`](https://github.com/deepset-ai/haystack/blob/main/haystack/core/component/sockets.py#L18) from which you can access all the [inputs info](https://github.com/deepset-ai/haystack/blob/main/haystack/core/component/types.py#L24) like the default values.

Reply:
Am I not getting the Component instance already from here:

`instance = self.graph.nodes[component_name]["instance"]`

?